File: Javascript/Basic Recursion/Problme7.js





function pallindrome(i,str){

    if(i>=Math.floor(str.length/2)) return true;
    
    else if(str[i]!==str[str.length-i-1]) return false;
    
  return  pallindrome(i+1,str);
    
  }
  
  
    
  
      
      
      function main(){
       let str="ama";
      // pallindrome(0,str)
     let bool=pallindrome(0,str);
     if(bool==true){
       console.log("pallindrome");
     }
     else{
       console.log("not pallindrome")
     }
      }
      
      
      main()
      

File: C++/Basic Recursion/Problme7.cpp

#include <iostream>
using namespace std;


bool pallindrome(int i, string& s){
  if(i>=s.length()/2) return true;
  else if(s[i]!=s[s.length()-i-1]) return false;
  
  pallindrome(i+1,s);
}

int main() 
{
   string s="ama";
  int val= pallindrome(0,s);
if(val==1){
  std::cout <<"pallindrome" << std::endl;
}
else{
  cout<<"not pallindrome";
}
    return 0;
}